### PR TITLE
Order Detail: Updated Payment section

### PR DESCRIPTION
**Pull request by pmusolino**
This PR is related to this issue: https://github.com/woocommerce/woocommerce-ios/issues/1090

## Tasks
- [x] Change "Fulfill Order" button to say "Begin Fulfillment"
- [x] Change "Subtotal" to "Product Total" and "Total" "Order Total"
- [x] Change payment details layout ("Paid by Customer" section)

## Changes
- Changed buttons and labels as specified
- Edited `PaymentTableViewCell` to match the new design
- `OrderPaymentDetailsViewModel` now support the new logic for the "Paid By Customer" section
- Implemented new tests for `PaymentTableViewCell` and `OrderPaymentDetailsViewModel`

## Testing

- Look at the code
- Run the app, and open two different order details: the first one with an awaiting payment, the second one with a payment sent
- Run WooCommerceTests

## Screenshots

| Awaiting payment             |  Payment sent |
:-------------------------:|:-------------------------:
Screen 1 | Screen 2

### CODE REVIEW

**Comment by pmusolino** [comment]
Ready for review @ctarda 👍

File: WooCommerce/Classes/ViewModels/OrderDetailsViewModel/OrderPaymentDetailsViewModel.swift [hunk 1]
```diff
@@ -51,17 +51,33 @@ final class OrderPaymentDetailsViewModel {
...
         }
 
+        if order.datePaid == nil {
+            return NSLocalizedString(
+                "Awaiting payment via \(order.paymentMethodTitle)",
+                comment: "Awaiting payment via (payment method title)"
+            )
+        }
+
+        let datePaid = order.datePaid!.toString(dateStyle: .medium, timeStyle: .none)
```
Comment (ctarda) [inline WooCommerce/Classes/ViewModels/OrderDetailsViewModel/OrderPaymentDetailsViewModel.swift]: I understand this is safe, as the code would have returned with "Awaiting payment..." if datePaid is nil, but I would prefer avoiding force unwraps (except in the tests). You never know when this line is going to be moved to a new placement, or the previous code that makes it safe suddenly disappears.

File: WooCommerce/WooCommerceTests/ViewRelated/PaymentTableViewCellTests.swift [hunk 2]
```diff
@@ -75,21 +75,34 @@ final class PaymentTableViewCellTests: XCTestCase {
         XCTAssertEqual(label.text, viewModel.totalValue)
     }
 
-    func testFooterTextContainsExpectedText() {
-        XCTAssertEqual(cell.getFooterText(), viewModel.paymentSummary)
+    func testPaidByCustomerContainsExpectedText() {
```
Comment (ctarda) [inline WooCommerce/WooCommerceTests/ViewRelated/PaymentTableViewCellTests.swift]: Nice!

File: WooCommerce/Classes/ViewModels/OrderDetailsViewModel/OrderPaymentDetailsViewModel.swift (hunk 1 above)
Comment (pmusolino) [reply WooCommerce/Classes/ViewModels/OrderDetailsViewModel/OrderPaymentDetailsViewModel.swift]: That makes a lot of sense, I will remember it for the future. 👍

**Comment by pmusolino** [comment]
Ready for a new review @ctarda 👍

**Comment by ctarda** [comment]
@kyleaparker @pmusolino mentioned you have already 👍 'ed these changes, so I'll merge now.

**Pull request merged**